index.bs

```diff
@@ -9847,49 +9847,52 @@ Methods</h5>
 				result of
 				<code><a href="http://www.ecma-international.org/ecma-262/6.0/#sec-get-o-p">Get</a>(O=<i>processorCtor</i>, P="parameterDescriptors")</code>.
 
-			1. Let <var>parameterDescriptorSequence</var>
-				be the result of
-				<a href="https://heycam.github.io/webidl/#es-to-sequence">
-				the conversion</a> from
-				<var>parameterDescriptorsValue</var>
-				to an IDL value of type
-				<code>sequence&lt;AudioParamDescriptor&gt;</code>.
-
-			1. Let <var>paramNames</var> be an empty Array.
-
-			1. For each <var>descriptor</var> of
-				<var>parameterDescriptorSequence</var>:
-
-				1. Let <var>paramName</var> be the value of
-					the member {{AudioParamDescriptor/name}}
-					in <var>descriptor</var>. Throw
-					a {{NotSupportedError}} if
-					<var>paramNames</var> already
-					contains <var>paramName</var> value.
-
-				1. Append <var>paramName</var> to
-					the <var>paramNames</var> array.
-
-				1. Let <var>defaultValue</var> be the value of
-					the member
-					{{AudioParamDescriptor/defaultValue}}
-					in <var>descriptor</var>.
-
-				1. Let <var>minValue</var> be the value of
-					the member
-					{{AudioParamDescriptor/minValue}}
-					in <var>descriptor</var>.
-
-				1. Let <var>maxValue</var> be the value of
-					the member
-					{{AudioParamDescriptor/maxValue}}
-					in <var>descriptor</var>.
-
-				1. If <var>defaultValue</var> is less than
-					<var>minValue</var> or greater than
-					<var>maxValue</var>,
-					<span class="synchronous">throw a
-					{{InvalidStateError}}</span>.
+			1. If <var>parameterDescriptorsValue</var> is not <code>undefined</code>,
```

Comment:
Can we not use double negative?

How about "exists"?  WebIDL seems to use this term when it checks the existence of things.

Reply:
`parameterDescriptorsValue` is a JS value and so WebIDL language is not helpful before conversion to a sequence, and `undefined` cannot be converted to a sequence.  ECMAScript has [`HasProperty()`](http://www.ecma-international.org/ecma-262/6.0/#sec-hasproperty), but that doesn't win us much because the property may be exist and still have the `undefined` value.

`undefined` is a JS value.  If that's what we choose to accept, then I assume we need to use that term.

Reply:
Not using a `not` here forces to use a jump, so I prefer to keep it.

I've had another look at this:

1. If the property doesn't exist, it's fine, then the value is `undefined`, the `AudioWorkletProcessor` simply has no parameter (see [0])
2. If it exists and returns `undefined` then it is accepted, and the `AudioWorkletProcessor` similarly has no parameter (see [1])
3. If something else than `undefined` is returned, an attempt to convert it to an iterable happens. In case of success, the other checks are performed. Otherwise and exception is thrown as part of the normal iterable-to-sequence algorithm. 

There is a difference between 1. and 2. as Karl notes, but distinguishing here is not particularly helpful, is it?

[0]: http://www.ecma-international.org/ecma-262/6.0/#sec-get-o-p Assert 1 passes, assert 2 passes, http://www.ecma-international.org/ecma-262/6.0/#sec-ordinary-object-internal-methods-and-internal-slots-get-p-receiver is called, assert 1 still passes, then it goes 2, skip 3, goes into 4, and then depending, returns `undefined` now or later, depending on the prototype chain
[1]: http://www.ecma-international.org/ecma-262/6.0/#sec-get-o-p Assert 1 passes, assert 2 passes, http://www.ecma-international.org/ecma-262/6.0/#sec-ordinary-object-internal-methods-and-internal-slots-get-p-receiver is called, assert 1 still passes, then it goes 2, skip 3, skip 4, skip 5, execute 6, skip 7, execute 8, this returns `undefined`.

Reply:
Yes. It makes sense. Thanks for the explanation. LGTM